functions/src/notifications/sendNotifications.ts

```diff
@@ -19,50 +22,69 @@ export const sendNotifications = async (
     )
+    const pushDestinations = destinations.filter(
+        (dest) => dest.topic && dest.topic.length > 4
+    )
 
     const emailsSent: string[] = []
     const smssSent: string[] = []
+    const pushsSent: string[] = []
 
     const appEnv = getAppEnv()
 
-    if (emailDestinations.length > 0) {
-        for (const emailDest of emailDestinations) {
-            if (!emailDest.email) continue
-            try {
-                await sendNotification({
-                    ...notificationContent,
-                    type: NotificationType.EmailNotificationType,
-                    lang: emailDest.lang,
-                    email: emailDest.email,
-                    emailFrom: appEnv.emailFrom,
-                })
-                emailsSent.push(emailDest.email)
-            } catch (error) {
-                console.error(error)
-            }
+    for (const emailDest of emailDestinations) {
+        if (!emailDest.email) continue
+        try {
+            await sendNotification({
+                ...notificationContent,
+                type: NotificationType.EmailNotificationType,
+                lang: emailDest.lang,
+                email: emailDest.email,
+                emailFrom: appEnv.emailFrom,
+            })
+            emailsSent.push(emailDest.email)
+        } catch (error) {
+            console.error(error)
+        }
+    }
+
+    for (const smsDest of smsDestinations) {
+        if (!smsDest.phone) continue
+        try {
+            await sendNotification({
+                ...notificationContent,
+                type: NotificationType.SmsNotificationType,
+                country: smsDest.country,
+                lang: smsDest.lang,
+                phone: smsDest.phone,
+            })
+            smssSent.push(smsDest.phone)
+        } catch (error) {
+            console.error(error)
         }
     }
 
-    if (smsDestinations.length > 0) {
-        for (const smsDest of smsDestinations) {
-            if (!smsDest.phone) continue
-            try {
-                await sendNotification({
-                    ...notificationContent,
-                    type: NotificationType.SmsNotificationType,
-                    country: smsDest.country,
-                    lang: smsDest.lang,
-                    phone: smsDest.phone,
-                })
-                smssSent.push(smsDest.phone)
-            } catch (error) {
-                console.error(error)
-            }
+    for (const pushDest of pushDestinations) {
+        if (!pushDest.topic) continue
+        try {
+            await sendNotification({
+                ...notificationContent,
+                type: NotificationType.PushNotificationType,
+                lang: pushDest.lang,
+                topic: pushDest.topic,
+                additionalData: {
+                    roomId: roomId,
+                },
+            })
+            pushsSent.push(pushDest.topic)
+        } catch (error) {
+            console.error(error)
```

Comment:
pls consider moving the 3 for loops into dedicated functions, and to try to refactor a little in order to avoid duplication

Reply:
done, 115 lines vs 90 :p